nimbus/src/lib.rs

```diff
@@ -80,7 +82,23 @@ impl NimbusClient {
             .map(|e| e.branch_slug.clone()))
     }
 
+    pub fn get_global_user_participation(&self) -> Result<Option<bool>> {
+        self.db()?.get(StoreId::Meta, DB_KEY_GLOBAL_USER_PARTICIPATION)
+    }
+
+    pub fn set_global_user_participation(&self, flag: bool) -> Result<()> {
+        self.db()?.put(StoreId::Meta, DB_KEY_GLOBAL_USER_PARTICIPATION, &flag)
+    }
+
+    fn is_enabled(&self) -> Result<bool> {
+        let user_choice = self.get_global_user_participation()?;
+        Ok(user_choice.unwrap_or(DEFAULT_GLOBAL_USER_PARTICIPATION))
+    }
+
     pub fn get_active_experiments(&self) -> Result<Vec<EnrolledExperiment>> {
+        if !self.is_enabled()? {
+            return Ok(Default::default());
+        }
```

Comment:
Doing the check here means we won't report membership in `get_experiment_branch`.

We might also want to prevent the `update_experiments` to happen too.